### ISSUE TITLE
[FEATURE REQ] Add support to `DefaultAzureCredential` authentication via `Azure.Identity` library

### ISSUE THREAD

**Issue by ilya-git** [Azure/azure-sdk-for-net#23770]
**Library or service name.**
Microsoft.Azure.WebJobs.Extensions.ServiceBus
 
**Is your feature request related to a problem? Please describe.**
Right now the connection string supports Managed Identity via:

`MyConnectionString__fullyQualifiedNamespace`

This is very nice and useful, but the problem is that for local development I would still have to revert to SharedAccessKey and cannot use e.g. `Azure CLI` or `Environment variables` authentication that `Azure.Identity` provides via `DefaultAzureCredential`.

It would be very nice if we can have the full power of `DefaultAzureCredential` via some option and not only `ManagedIdentity` so that we won't need to supply secrets like shared access key for running locally. Ordinary service bus client library already supports it.

P.S. I have already assumed that this was the case and maybe it is, but I did not manage to get it working locally and created the bug [here](https://github.com/Azure/azure-sdk-for-net/issues/23768). The bug can be just closed if this is indeed something not supported yet (hopefully there will be no warning if this feature is implemented)

**Comment by jsquire** [comment]
//fyi: @pakrym

**Comment by jongio** [comment]
Please see the beta version here which supports Azure Identity
https://www.nuget.org/packages/Microsoft.Azure.WebJobs.Extensions.ServiceBus/5.0.0-beta.5

Blog here:
https://devblogs.microsoft.com/azure-sdk/introducing-the-new-azure-function-extension-libraries-beta/

Working code here: 
https://github.com/jongio/memealyzer/tree/main/src/net/Services/Functions

**Comment by ilya-git** [comment]
I have tried locally with MyConnectionString__fullyQualifiedNamespace and did not quite work for me locally with environment variables. It does work with Managed Identity in Azure.

I know that the environment variables are set correctly since I use them in other places that do work. Perhaps I did something else wrong. Is it enough to just specify this value in local.settings.json?

**Comment by pakrym** [comment]
Do you by any chance have the MyConnectionString__credential set?

**Comment by JoshLove-msft** [comment]
When you say it does not work locally, do you mean that you see the warning on startup that you referenced in the linked issue? This is a known issue with the core tools that will be updated.
Apart from that, does the function still work?

**Comment by ilya-git** [comment]
No, I got an error that I did not have enough rights to connect the service bus. I did not have `MyConnectionString__credential` set, only 3 env variables to represent a service principal. Maybe I did something wrong with the access, though I set it on the subscription. If you say that this feature is implemented I can try again on Monday to verify once again. I can report then exactly which error did happen.

**Comment by ilya-git** [comment]
Thanks a lot for help! 

It seems that it is working indeed as you described. I think I have made a rookie mistake and forgot to update the library, the last time I tried to test this. And then due to the fact that there was a warning and that documentation (unlike the referenced blogpost by @jongio) said it is only supported for Managed Identity I gave up and created this issue instead.

So far everything seems to work as expected, is it possible to get any estimates for the production version of 5.0.0 package or it's not something that you guys can share?
